### ISSUE TITLE
missing furnitures in some rooms

### ISSUE THREAD

**Issue by ghost** [Varkalandar/imp_city#8]
In some cases, furnitures are not placed, I do not know why (2nd screenshot shows another problem).

![missing-slot-impcity](https://github.com/Varkalandar/imp_city/assets/1316300/785b901f-3703-4ff6-aedc-250aa9e5fa6e)
![3bugs-impcity](https://github.com/Varkalandar/imp_city/assets/1316300/be94762f-9453-4b88-bcab-5ec723c1fd7b)

**Comment by Varkalandar** [comment]
I see. As mentioned in issue #9, I'll look into both problems. Both times the new room furnishing code seems to be faulty.

**Comment by Varkalandar** [comment]
Should be fixed in v0.26. Changing the room (by adding or removing a square) should set up furniture and lights properly and fix the problems present.

**Comment by ghost** [comment]
![2023-11-14_000](https://github.com/Varkalandar/imp_city/assets/1316300/afeeea47-57ad-4d01-ad97-c9f3e7847fcf)

Still happens, sorry. This is random though, I don't know how to reproduce.

**Comment by ghost** [comment]
More screenshots and a savegame.
Apparently, it is possible to fix the problem by destroying/rebuilding the _affected_ tiles, doing that with unaffected tiles of the same room does not change anything.

![2023-11-14_004](https://github.com/Varkalandar/imp_city/assets/1316300/9d1afc04-f7ea-4460-aa4f-573b19ba7b72)
![2023-11-14_003](https://github.com/Varkalandar/imp_city/assets/1316300/a4324bd8-ad5a-4c0d-9c4f-00e7c82a9601)
![2023-11-14_002](https://github.com/Varkalandar/imp_city/assets/1316300/9669d511-2212-4fd6-848a-d3c787f1c517)
[save.zip](https://github.com/Varkalandar/imp_city/files/13357041/save.zip)

**Comment by Varkalandar** [comment]
I see. At least it seems in v0.26 the problem is nit as severe anymore as it was in 0.25.

Thank you for the save game. I'll try to investigate the problem tomorrow.

**Comment by Varkalandar** [comment]
I think I could fix another few causes of this problem. The v0.26.1 at least solves the problems demonstrated this saved game, but just when I merged the code into my 0.27 dev branch, I noticed another mistake. Yet the 0.26.1 was already published. 

I'm sorry. The mentioned mistake is fixed on the 0.27 dev branch and should not hurt unless you do large scale deletion of room squares. Saving and loading the game should even fix the data structures in such a case. 

There is one known problem remaining though. If you cut an existing room into two by deleting squares, the game will still consider both areas as one room until you save and load again, after that it will be two separate rooms. I'm working on a solution for that and plan to publish it with v0.27
